### PR TITLE
Lionel Smith-Gordon Phase 2 of Asset Refactoring

### PR DESCRIPTION
**Pull request by regnosys-prod-user**

### CODE REVIEW

**Review by dshoneisda — DISMISSED** [review]
Reviewed at CRWG on 2nd July and approved on that basis

File: rosetta-source/src/main/rosetta/observable-asset-type.rosetta [hunk 1]
```diff
@@ -99,21 +152,38 @@ type Price extends PriceSchedule: <"Specifies a price as a single value to be as
...
+//         [metadata location]
+//     identifier AssetIdentifier (0..*) <"Comprises of an identifier and a symbology source. The associated metadata key denotes the ability to associate a hash value to the AssetIdentifier instantiations for the purpose of model cross-referencing, in support of functionality such as the event effect and the lineage.  This attribute will replace ProductIdentifier in the full release.">
+//         [metadata location]   
+//     currencyPair QuotedCurrencyPair (0..1) <"Describes the composition of a rate that has been quoted or is to be quoted, including the two currencies and the quotation relationship between the two currencies.">
+//         [metadata location]
+//     optionReferenceType OptionReferenceTypeEnum (0..1) <"The underlying contract which is referenced when determining the final settlement price of the instrument. Eg. Rolling Front Month Future; Spot etc.">
```
Comment (lolabeis) [inline rosetta-source/src/main/rosetta/observable-asset-type.rosetta:168]: Don't you need to maintain this `optionReferenceType` attribute in the new `Observable`?

File: rosetta-source/src/main/java/cdm/product/template/processor/CommodityClassificationMetaMappingProcessor.java [hunk 2]
```diff
@@ -25,12 +25,6 @@ public void map(Path synonymPath, List<? extends RosettaModelObjectBuilder> buil
         if (!forwardPayoutBuilders.isEmpty()) {
             ForwardPayout.ForwardPayoutBuilder forwardPayoutBuilder = forwardPayoutBuilders.get(0);
             // create reference
```
Comment (chrisisla) [inline rosetta-source/src/main/java/cdm/product/template/processor/CommodityClassificationMetaMappingProcessor.java:27]: the code to create the reference has been removed so I assume this comment should be removed too?

File: rosetta-source/src/main/rosetta/base-staticdata-asset-common-type.rosetta [hunk 3]
```diff
@@ -8,25 +8,40 @@ import cdm.base.staticdata.party.*
 import cdm.observable.asset.*
 import cdm.product.template.*
 
+choice Asset:  <"An Asset is defined something that can be owned and transferred in the financial markets. As a choice data type, one and only one of the  attributes must be used. Cash: An Asset that consists solely of a monetary holding in a currency. Commodity: An Asset comprised of raw or refined materials or agricultural products, eg gold, oil or wheat. DigitalAsset An Asset that exists only in digital form, eg Bitcoin or Ethereum; excludes the digital representation of other Assets. Instrument: An asset that is issued by one party to one or more others; Instrument is also a choice data type.">
```
Comment (chrisisla) [inline rosetta-source/src/main/rosetta/base-staticdata-asset-common-type.rosetta:11]: having an odd feeling of deja vu with this comment... see comment I made on #3022

File: rosetta-source/src/main/rosetta/base-staticdata-asset-common-type.rosetta [hunk 4]
```diff
@@ -198,14 +227,18 @@ type CollateralTaxonomyValue: <"Specifies the collateral taxonomy value, either
...
-type Security extends ProductBase: <"Identifies a security by referencing a product identifier and by specifying the sector.">
+type Security extends InstrumentBase: <"Identifies a security by referencing a product identifier and by specifying the sector.">
     [docReference ICMA GMRA namingConvention "Purchased Security"
 		provision "As defined in GMRA paragraph 2(oo) The Purchased Securities are the Securities sold or to be sold and any New Purchased Securities transferred by Seller to Buyer under paragraph 8 (Substitution)."]
     securityType SecurityTypeEnum (1..1) <"Identifies the type of security using an enumerated list.">
     debtType DebtType (0..1) <"Identifies the type of debt and selected debt economics.">
     equityType EquityTypeEnum (0..1) <"Identifies the type of equity.">
     fundType FundProductTypeEnum (0..1) <"Identifies the type of fund.">
+
+    // temporary items
```
Comment (chrisisla) [inline rosetta-source/src/main/rosetta/base-staticdata-asset-common-type.rosetta:238]: Can you confirm when these temporary items will be addressed please? Is it to be done in the Phase 3 PR?

**Comment by Oblongs** [comment]
Superceded by #3039

File: rosetta-source/src/main/rosetta/observable-asset-type.rosetta (hunk 1 above)
Comment (Oblongs) [reply rosetta-source/src/main/rosetta/observable-asset-type.rosetta:168]: I am watching this and hoping not.  I think it is covered by the new ListedDerivative structure as this is not, IMO, an attribute of `Observable`.  The attribute is not referenced anywhere else in the CDM nor DRR

File: rosetta-source/src/main/rosetta/observable-asset-type.rosetta [hunk 5]
```diff
@@ -99,21 +152,38 @@ type Price extends PriceSchedule: <"Specifies a price as a single value to be as
...
+//         required choice rateOption, commodity, productIdentifier, currencyPair
+
+// ARTForce: opportunity to change to 'choice'
+type Observable:  <"Specifies the object to be observed for a price, it could be an asset or a reference to something in the market.">
+    asset Asset (0..1) <"An Asset is defined something that can be owned and transferred in the financial markets.">
+    basket Basket (0..1) <"Defines a custom basket by referencing an identifier and its consituents.">
+    currencyPair QuotedCurrencyPair (0..1)  <"The composition of a foreign exchange rate that has been quoted or is to be quoted.">
         [metadata location]
-    currencyPair QuotedCurrencyPair (0..1) <"Describes the composition of a rate that has been quoted or is to be quoted, including the two currencies and the quotation relationship between the two currencies.">
+    productIdentifier ProductIdentifier (0..1)  <"Comprises an identifier and a source of a Product.">
```
[the remark this answers is not in the document]
Comment (Oblongs) [reply rosetta-source/src/main/rosetta/observable-asset-type.rosetta:179]: Yes, planned for Phase 3; have added a comment to make this clear

File: rosetta-source/src/main/rosetta/observable-asset-type.rosetta [hunk 6]
```diff
@@ -99,21 +152,38 @@ type Price extends PriceSchedule: <"Specifies a price as a single value to be as
...
+
+// ARTForce: opportunity to change to 'choice'
+type Observable:  <"Specifies the object to be observed for a price, it could be an asset or a reference to something in the market.">
+    asset Asset (0..1) <"An Asset is defined something that can be owned and transferred in the financial markets.">
+    basket Basket (0..1) <"Defines a custom basket by referencing an identifier and its consituents.">
+    currencyPair QuotedCurrencyPair (0..1)  <"The composition of a foreign exchange rate that has been quoted or is to be quoted.">
         [metadata location]
-    currencyPair QuotedCurrencyPair (0..1) <"Describes the composition of a rate that has been quoted or is to be quoted, including the two currencies and the quotation relationship between the two currencies.">
+    productIdentifier ProductIdentifier (0..1)  <"Comprises an identifier and a source of a Product."> 
+    rateOption FloatingRateOption (0..1) <"Specification of a floating rate option as a floating rate index and tenor.">
```
[the remark this answers is not in the document]
Comment (Oblongs) [reply rosetta-source/src/main/rosetta/observable-asset-type.rosetta:180]: Yes, planned for Phase 3; have added a comment to make this clear

File: rosetta-source/src/main/rosetta/observable-asset-type.rosetta [hunk 7]
```diff
@@ -99,21 +152,38 @@ type Price extends PriceSchedule: <"Specifies a price as a single value to be as
...
         [metadata location]
-    optionReferenceType OptionReferenceTypeEnum (0..1) <"The underlying contract which is referenced when determining the final settlement price of the instrument. Eg. Rolling Front Month Future; Spot etc.">
+    index Index (0..1)  <"Defines the different types of Index.">
+        condition: one-of
 
-    condition ObservableChoice: <"An observable can only be composed of one type any time.">
-        required choice rateOption, commodity, productIdentifier, currencyPair
+type Basket extends AssetIdentifier: <"Defines a custom basket by referencing an identifier and its consituents.">
```
[the remark this answers is not in the document]
Comment (Oblongs) [reply rosetta-source/src/main/rosetta/observable-asset-type.rosetta:185]: Done now that other `Basket` changes have been incorporated in new P2 PR

File: rosetta-source/src/main/rosetta/product-template-type.rosetta [hunk 8]
```diff
@@ -23,11 +23,22 @@ import cdm.product.collateral.*
...
+type TransferableProduct extends ProductBase:  <"A TransferableProduct is a type of Product which can be used in a SettlementCommitment for a basic cash settled trade of either an Asset with or without the addition of specific EconomicTerms.">
+    asset Asset (1..1)
+
+type ProductBase: <"Serves as an abstract class to specify a product using a productIdentifier.">
+    productTaxonomy ProductTaxonomy (0..*) <"Specifies the product taxonomy, which is composed of a taxonomy value and a taxonomy source.">
+    economicTerms EconomicTerms (0..1)
+   
+type NewProduct extends ProductBase:
+    productIdentifier ProductIdentifier (0..*) <"Comprises an identifier and a source. The associated metadata key denotes the ability to associate a hash value to the ProductIdentifier instantiations for the purpose of model cross-referencing, in support of functionality such as the event effect and the lineage.">
+    //    [metadata address "pointsTo"=Observable->productIdentifier]
```
[the remark this answers is not in the document]
Comment (Oblongs) [reply rosetta-source/src/main/rosetta/product-template-type.rosetta:35]: was temporary addition, not in latest PR

File: rosetta-source/src/main/rosetta/product-template-type.rosetta [hunk 9]
```diff
@@ -81,6 +92,10 @@ type EconomicTerms: <" This class represents the full set of price-forming featu
         if payout -> assetPayout -> dividendTerms exists
         then terminationDate exists
 
+choice Underlier:
```
[the remark this answers is not in the document]
Comment (Oblongs) [reply rosetta-source/src/main/rosetta/product-template-type.rosetta:95]: updated in P2 PR

File: rosetta-source/src/main/rosetta/observable-asset-type.rosetta (hunk 1 above)
Comment (lolabeis) [reply rosetta-source/src/main/rosetta/observable-asset-type.rosetta:168]: Got it. That would be good news. In that case, delete the enum altogether.

Comment (Oblongs) [reply rosetta-source/src/main/rosetta/observable-asset-type.rosetta:168]: Done in new PR

**Pull request closed without merging**